sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/BeamSqlFnExecutor.java

```diff
@@ -504,21 +501,6 @@ private static BeamSqlExpression getBeamSqlExpression(RexNode rexNode) {
           ret = new BeamSqlIsNotNullExpression(subExps.get(0));
           break;
 
-        case "HOP":
-        case "TUMBLE":
-        case "SESSION":
-          ret = new BeamSqlWindowExpression(subExps, node.type.getSqlTypeName());
```

Comment:
This change seems disabling grouped auxiliary functions (HOP/TUMBLE/SESSION START/END). Will code generation add them back?

Reply:
It actually doesn't disable them. The implementation of `BeamSqlWindowExpression` just passes through the timestamp: https://github.com/apache/beam/pull/6382/files#diff-ec3df29f3fc44a5ade2f7f8ef33920ecL51
```
(ReadableInstant) operands.get(0).evaluate(inputRow, window, env).getValue());
```

 This is now happening directly at the project layer: https://github.com/apache/beam/pull/6382/files#diff-b0b2c6af2f694e5f6d9a1c64cbc482a1R81
```
projects.set(groupFieldIndex, rexCall.getOperands().get(0));
```

Reply:
SGTM

Reply:
It seems that our implementation of window start/end has never worked because Calcite convert window start to a column reference and window end to a PLUS operation.